docs/ref/forms/modelform-meta.txt

```diff
@@ -0,0 +1,283 @@
+
+    If no class is specified, a :exc:`ValueError` exception will be raised.
+
+
+.. _ref-modelform-meta-specifying-model-fields:
+
+Specifying which model fields to include
+========================================
+
+``fields``
+----------
+
+.. attribute:: ModelFormOptions.fields
+
+    *One of* :attr:`~ModelFormOptions.fields` *and* :attr:`~ModelFormOptions.exclude` *must be set.* Expects a tuple of string or list of strings where each string is the name of a field from the model set in :attr:`~ModelFormOptions.model` or, instead, the single string ``'__all__'``. Every field named in :attr:`~ModelFormOptions.fields` will be included in the output form. Non-editable fields cannot be included.
+    
+    * If set to the string ``'__all__'``, all non-editable model fields will be
+      inlcuded.
+    * Other than the special case of ``'__all__'``,
+      :attr:`~ModelFormOptions.fields` **cannot** be set to a single string,
+      even if only a single field is desired. The input for
+      :attr:`~ModelFormOptions.fields` must still be structured as a tuple of
+      strings (e.g., ``('myFieldName',)``) or list of strings (e.g., ``[
+      'myFieldName' ]``) or else a :exc:`TypeError` will be raised.
+    * If a string provided is not a valid field name for the model specified by
+      :attr:`ModelFormOptions.model` (e.g., the field doesn't exist on the model
+      or the field name was mistyped), a
+      :exc:`~django.core.exceptions.FieldError` exception will be raised.
+    * If a non-editable field name is provided, a
+      :exc:`~django.core.exceptions.FieldError` exception will be raised.
+    * If neither of :attr:`~ModelFormOptions.fields` or
+      :attr:`~ModelFormOptions.exclude` are set, an
+      :exc:`~django.core.exceptions.ImproperlyConfigured` exception will be
+      raised.
+
+``exclude``
+-----------
+
+.. attribute:: ModelFormOptions.exclude
+
+    *One of* :attr:`~ModelFormOptions.fields` *and* :attr:`~ModelFormOptions.exclude` *must be set.* Expects a tuple of strings or list of strings where each string is the name of a field from the model set in :attr:`~ModelFormOptions.model`. Every field named in :attr:`~ModelFormOptions.exclude` will be excluded from the output form, **including fields that are also listed in** :attr:`~ModelFormOptions.fields`.
+
+    * When using :attr:`~ModelFormOptions.exclude`, leaving
+      :attr:`~ModelFormOptions.fields` unset or setting it to ``None`` has the
+      net effect of including all valid model fields (as if
+      :attr:`~ModelFormOptions.fields` were set to ``'__all__'``) except for
+      those provided to :attr:`~ModelFormOptions.exclude`. 
+    * If neither of :attr:`~ModelFormOptions.fields` or
+      :attr:`~ModelFormOptions.exclude` are set, an
+      :class:`~django.core.exceptions.ImproperlyConfigured` exception will be
+      raised.
+
+.. _ref-modelform-meta-controlling-model-field-behavior:
+
+Controlling how included fields behave
```

Comment:
I think this heading looks like it belongs to the topic on ModelForms 🤔 , what do you think. I also think that it breaks the order of this reference, which is that we are expecting a list of options/attributes and their descriptions.

Reply:
I noticed that `modelform_factory` arguments closely match `ModelForm`'s `Meta` attributes.

Reply:
Yes, it's a bit of a weird one. My goal was to segment the options in this reference entry by mandatory and (somewhat) required fields, and then, well, by all the rest (alphabetically). I ended up labelling the sections by function, but descriptive headings, such as "Required `Meta` attributes" and then "Optional `Meta` attributes" might work better for the tone of a reference entry.